### ISSUE TITLE
Name is overwritten after restart homebridge

### ISSUE THREAD

**Issue by Nastras** [MiguelRipoll23/homebridge-securitysystem#462]
### Plugin version

7.5.1

### Plugin configuration

{
"bridge": {
"name": "homebridge-securitysystem",
"username": "",
"port": ,
"pin": "",
"manufacturer": "",
"model": "Ak-SHS-2",
"advertiser": "avahi",
"bind": ["eth0"]
},

"plugins": [
"homebridge-securitysystem"
],

"accessories": [
{
"accessory": "security-system",
"name": "Wh Alarmanlage",
"default_mode": "off",
"disabled_modes": ["Home"],
"trip_switch_name": "Alarmsirene",
"siren_switch": true,
"siren_mode_switches": false,
"override_off": true,
"save_state": true,
"test_mode": false,
"arm_seconds": 0,
"trigger_seconds": 0,
"reset_minutes": 3,
"reset_off": true,
"command": true,
"command_current_home": "",
"command_current_away": "",
"command_current_night": "",
"command_current_off": "",
"command_current_triggered":""
},
{
"accessory": "security-system",
"name": "Gr Alarmanlage",
"default_mode": "off",
"disabled_modes": ["Home"],
"trip_switch_name": "Alarmsirene",
"siren_switch": true,
"siren_mode_switches": false,
"override_off": true,
"save_state": true,
"test_mode": false,
"arm_seconds": 0,
"trigger_seconds": 0,
"reset_minutes": 3,
"reset_off": true,
"command": true,
"command_current_home": "",
"command_current_away": "",
"command_current_night": "",
"command_current_off": "",
"command_current_triggered":""
}
]
}

### Homebridge version

1.6.1

### What's your question?

Hello @MiguelRipoll23 

I have the following problem. The name of the switch alarm is overwritten in Homekit after each restart of the Homebridge to the value that is in the config.json. Do you have an idea where the problem is?

![IMG_4556](https://github.com/MiguelRipoll23/homebridge-securitysystem/assets/27686185/a631078f-ee60-410f-86e8-372279b7d120)
![IMG_0081](https://github.com/MiguelRipoll23/homebridge-securitysystem/assets/27686185/e617a8c3-0bca-4da6-b421-7e6a6584ddd3)
![IMG_0082](https://github.com/MiguelRipoll23/homebridge-securitysystem/assets/27686185/8932e671-6d27-4ab5-af37-2ed25ff1e095)

**Comment by MiguelRipoll23** [comment]
Hey there,

To fully understand the issue as I'm not familiar with the app you're using I've some questions:

- Is the name issue related with the Security system's accessory? (Not the trip switch or other other switches exposed by the plugin)
- Have you changed the name using a HomeKit app? (Outside Homebridge configuration)

I've seen issues happening with the configuration when multiple instances of the plugin are used like your case, you can also try to use a child bridge (a Homebridge feature) to prevent conflict between other instances.

**Comment by Nastras** [comment]
Hey @MiguelRipoll23 thanks for the quick answer 👌

`Is the name issue related with the Security system's accessory? (Not the trip switch or other other switches exposed by the plugin)
`
It is the device name of the alarm system. It is not a custom feature. If I change the name from "Wh Alarm System" to "Alarm System" in Homekit, this change resets after a restart of Homebridge.
The app I used here is called HomeDash but the behavior is the same when I change it in the Eve app.

`I've seen issues happening with the configuration when multiple instances of the plugin are used like your case, you can also try to use a child bridge (a Homebridge feature) to prevent conflict between other instances. `

What error do you mean exactly? So far everything works as expected except that the name change is not applied.

If you say that it is better to use a second instance I can of course set that up?

**Comment by Nastras** [comment]
Here a screen from Eve App:

![IMG_0084](https://github.com/MiguelRipoll23/homebridge-securitysystem/assets/27686185/31c2d23f-0862-4e91-b52a-f291df91bd0b)
![IMG_0083](https://github.com/MiguelRipoll23/homebridge-securitysystem/assets/27686185/61833cb5-a13d-47d6-9a73-bd8edad71a76)

**Comment by MiguelRipoll23** [comment]
I fully understand the issue you're having right now. Unfortunately changes to accessory names that are applied outside Homebridge like for example an iOS/macOS device won't be persisted.

The new name should be applied to the Homebridge configuration in order to avoid the name being reset. While it's possible to receive name changes coming from HomeKit it would add more complexity into the plugin.

With respect to the multiple instances issue,see #346 for example. It happens to other options mostly.

**Comment by Nastras** [comment]
`I fully understand the issue you're having right now. Unfortunately changes to accessory names that are applied outside Homebridge like for example an iOS/macOS device won't be persisted.

The new name should be applied to the Homebridge configuration in order to avoid the name being reset. While it's possible to receive name changes coming from HomeKit it would add more complexity into the plugin.`

Good that you understand my question, but I understand your answer only partially 😅. Sorry.

So to summarize again:

I assign in the config.json a device name
"name": "Wh Alarm", if I change this name in homekit to "Alarm" the name in homekit will be changed again to " Wh Alarm" after a restart of the homebridge.

With every other plugin homekit remembers the changed device name but not here. It looks like the plugin always reports the name from config.json to homekit again.

Can the behavior possibly be a bug that is solved by a second instance?

**Comment by MiguelRipoll23** [comment]
> I assign in the config.json a device name
"name": "Wh Alarm", if I change this name in homekit to "Alarm" the name in homekit will be changed again to " Wh Alarm" after a restart of the homebridge.

This is the expected behaviour at the moment. When you change an accessory name an event is sent to Homebridge, if the plugin has implemented an event receiver for accessory name changes it can read the new name and persist it in a file. Although the name from the configuration will still be outdated.

The plugin doesn't have implemented at the moment the receiver neither the logic to handle this kind of event for the accessories exposed from the plugin.

>  With every other plugin homekit remembers the changed device name but not here. It looks like the plugin always reports the name from config.json to homekit again.

Could you provide me some plugin examples? Maybe if I find an easier way to implement this functionality in can land in an upcoming release.

> Can the behavior possibly be a bug that is solved by a second instance?

Nope! It does happen with one instance as well.

**Comment by Nastras** [comment]
`Could you provide me some plugin examples? Maybe if I find an easier way to implement this functionality it can land in an upcoming release.`

https://github.com/ebaauw/homebridge-hue
https://github.com/valiquette/homebridge-easee
https://github.com/hans-1/homebridge-cmdtriggerswitch

`This is the expected behaviour at the moment. When you change an accessory name an event is sent to Homebridge, if the plugin has implemented an event receiver for accessory name changes it can read the new name and persist it in a file. Although the name from the configuration will still be outdated.

The plugin doesn't have implemented at the moment the receiver neither the logic to handle this kind of event for the accessories exposed from the plugin.`

Thank you for the explanation 👍.

**Comment by MiguelRipoll23** [comment]
Those plugins don’t have any kind of functionality to receive updated accessory names, probably depends on the HomeKit data from the device.

The reason it doesn’t work for the security system plugin is because it exposes an accessory instead of a platform and sub-accessories instead of a single accessory.

I will have to research on this one to see if something is possible to implement it easily.

**Comment by Nastras** [comment]
Sorry. I assumed it was available in one of the plugins. 
Thanks for your effort!

**Comment by MiguelRipoll23** [comment]
Plugin has been rewritten to a dynamic plugin, this issue won’t affect anymore after the upgrade is done 😄
